src/codegen/expression.rs

```diff
@@ -2552,12 +2572,25 @@ fn array_subscript(
                         array_length
                     }
                 } else {
-                    Expression::Builtin(
+                    //if a subscript is encountered array length will be called
+
+                    //return array length by default
+                    let mut returned = Expression::Builtin(
                         *loc,
                         vec![Type::Uint(32)],
                         Builtin::ArrayLength,
                         vec![array.clone()],
-                    )
+                    );
+
+                    if let Expression::Variable(loc, _ty, num) = array.clone() {
+                        // If the size is known aka in our cfg.array_length_map, do the replacement
+
+                        if let Some(array_size_unrwapped) = cfg.array_lengths_temps.get(&num) {
```

Comment:
```suggestion
                        if let Some(array_size_unrwapped) = cfg.array_lengths_temps.get(num) {
```
This works if you do not clone `array`.